src/main/java/io/github/fablabsmc/fablabs/impl/fiber/annotation/AnnotatedSettingsImpl.java

```diff
@@ -319,14 +320,59 @@ private static void checkViolation(Field field) throws FiberException {
 
 		try {
 			field.setAccessible(true);
-			this.applyToNode(sub, field.get(pojo));
+			Object group = field.get(pojo);
+
+			// If group is null, let's try creating a new instance ourselves:
+			if (group == null) {
+				group = tryInstantiateClass(pojo, field.getType());
+				// sets the field so the POJO receives the object we made
+				field.set(pojo, group);
+			}
+
+			this.applyToNode(sub, group);
 			this.applyAnnotationProcessors(pojo, field, sub, this.groupSettingProcessors);
 			sub.build();
 		} catch (IllegalAccessException e) {
 			throw new FiberException("Couldn't fork and apply sub-node", e);
 		}
 	}
 
+	private <P> Object tryInstantiateClass(P pojo, Class<?> clazz) throws FiberException {
+		Constructor<?> constructor = null;
+		boolean accessible = false;
+
+		try {
+			// Inner classes have an extra parameter in their constructor: a reference to the 'upper class'.
+			boolean isInner = !Modifier.isStatic(clazz.getModifiers());
+
+			if (isInner) {
+				constructor = Arrays.stream(clazz.getDeclaredConstructors())
+						.filter(c -> c.getParameterCount() == 1)
+						.findAny()
+						.orElseThrow(() -> new FiberException("Couldn't create group from inner class " + clazz.getSimpleName() + ", is the class public and does it have a nullary constructor?"));
+			} else { // static nested class
+				constructor = clazz.getDeclaredConstructor();
+			}
+
+			accessible = constructor.isAccessible();
+			constructor.setAccessible(true);
+
+			return isInner ? constructor.newInstance(pojo) : constructor.newInstance();
+		} catch (InstantiationException e) {
+			throw new FiberException("Couldn't create group from class " + clazz.getSimpleName() + ". It may not be an interface or an abstract class, and it must have a nullary constructor.", e);
+		} catch (InvocationTargetException e) {
+			throw new FiberException("Couldn't create group from class " + clazz.getSimpleName(), e);
```

Comment:
I recommend unwrapping the `InvocationTargetException` and setting the cause based on the unwrapped exception

Reply:
Are there any possible underlying exceptions we can predict? From the javadocs, `InvocationTargetException` is only thrown when the constructor throws the exception, which I suppose could only happen if the user wrote their own nullary constructor & throw an exception themselves?

Reply:
I mean throw `FiberException(..., e.getCause())`